org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/server/ServerModule.xtend

```diff
@@ -28,9 +28,8 @@ class ServerModule extends AbstractModule {
 		
     	bind(LanguageServer).to(LanguageServerImpl)
         bind(IResourceServiceProvider.Registry).toProvider(ResourceServiceProviderServiceLoader)
-        bind(IWorkspaceConfigFactory).to(ProjectWorkspaceConfigFactory)
+        bind(IWorkspaceConfigFactory).to(MultiRootWorkspaceConfigFactory)
```

Comment:
Clients that used to have bound `ProjectWorkspaceConfigFactory` to their special subtype would now be broken. If MultiRootWorkspaceConfigFactory was a subtype of ProjectWorkspaceConfigFactory, that could be mitigated.

Reply:
All right. Even though `MultiRootWorkspaceConfigFactory` will not use or expose any method of `ProjectWorkspaceConfigFactory`.